types.d.ts

```diff
@@ -36,8 +41,10 @@ declare module "@nice-digital/global-nav" {
 		search?: false | SearchProps;
 		cookie?: boolean;
 		auth?: {
-			provider?: "niceAccounts";
-			environment?: "live" | "beta" | "test" | "local";
+			provider?: string;
```

Comment:
I think we can be a bit more explicit with the TS defs and use a union type to distinguish between the allowed props. e.g.:

```
type IDAMProviderProps = {
provider: "idam;
links?: Array<Links>;
			displayName?: string;
};
type AccountsProviderProps = {
provider?: "niceAccounts";
environment?: "live" | "beta" | "test" | "local";
}

...
auth?: AccountsProviderProps | IDAMProviderProps;
```